File: My Graph/My Graph/ANode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace My_Graph
{
    class ANode
    {
        private char m_ID;
        private int data;
        private bool m_visited;
        private List<AEdge> m_Edges = new List<AEdge>();
        
        public ANode(char ID, int dada)
        { 
        
        }

        public void AddEdge(char cEnd, int weight)
        {
        AEdge edge = new AEdge((m_ID,cEnd,weight);
        m_Edges.Add(edge)
        }

        public void DeleteEdge(char cEnd)
        {
        //AEdge edge = new AEdge((m_ID,cEnd,weight);
        
            for(int i=0; i<m_Edges.Count();i++)
            {
                AEdge edge = m_Edges(i);
                if (edge.getEnd == cEnd)
                {
                    m_Edges.erase(m_Edges.begin() + i);
                    break;
                }

            }
        }

    }
}

File: Graph/GraphAdjList/ANode.cpp
#include "stdafx.h"
#include "ANode.h"

ANode::ANode(char ID, int data)
{
	m_ID = ID;
	m_data = data;
	m_Visited = false;
}

ANode::~ANode()
{
	// delete the node 
	// delete all the edges
	for(int i = 0 ; i < m_edges.size(); i ++)
	{
		AEdge *edge = 	m_edges.back();
		delete edge;
		m_edges.pop_back();
	}
}

void ANode::AddEdge(char cEnd, int weight)
{
	AEdge *edge = new AEdge(m_ID, cEnd, weight);
	m_edges.push_back(edge);
}
void ANode::DeleteEdge(char cEnd)
{
	for(int i = 0 ; i < m_edges.size(); i ++)
	{
		AEdge *edge = m_edges.at(i);
		if(edge->GetEnd() == cEnd)
		{
			m_edges.erase(m_edges.begin() + i);
			break;
		}
	}
}

char ANode::GetID()
{
	return m_ID;
}


std::vector<char> ANode::GetNeighbours()
{
	std::vector<char> neighbours;
	for(int i = 0 ; i < m_edges.size(); i ++)
	{
		neighbours.push_back(m_edges[i]->GetEnd());
	}
	return neighbours;
}

bool ANode::IsVisited()
{
	return m_Visited;
}
void ANode::SetVisited(bool visited)
{
	m_Visited = visited;
}

int ANode::GetNumEdges()
{
	return m_edges.size();
}